test-app/runtime/src/main/cpp/MetadataNode.cpp

```diff
@@ -1785,8 +1790,37 @@ void MetadataNode::BuildMetadata(const string& filesPath) {
     baseDir.append("/metadata");
 
     DIR* dir = opendir(baseDir.c_str());
+
     if(dir == nullptr){
-        throw NativeScriptException(string("metadata folder couldn't be opened!"));
+        stringstream ss;
+        ss << "metadata folder couldn't be opened! (Error: ";
+        ss << errno;
+        ss << ") ";
+
+        // TODO: Is there a way to detect if the screen is locked as verification
+        // We assume based on the error that this is the only way to get this specific error here at this point
+        if (errno == 2) {
```

Comment:
Isnt it better to use here the "ENOENT" constant out of errno.h instead of compare against 2 ? :)

Reply:
Yes, it would be much better.  Thanks for the heads up.    You should have seen all the testing code I chopped out...  I missed cleaning that one up.  :grinning:

Reply:
Fixed in new Push.